packages/shared/scope-hoisting/src/link.js

```diff
@@ -630,35 +626,33 @@ export function link({
           return;
         }
 
-        let {parent} = path;
-        // Check if $id$export$name exists and if so, replace the node by it.
-        if (isAssignmentExpression(parent)) {
+        let {parent, parentPath} = path;
+        // If inside an expression, update the actual export binding as well
+        if (isAssignmentExpression(parent, {left: path.node})) {
```

Comment:
We never actually checked if the member expression is on the LHS....
So `output = value.foo` actually matched